foundationdb/datadog_checks/foundationdb/check.py

```diff
@@ -84,28 +173,35 @@ def check_metrics(self, status):
         if "workload" in cluster:
             workload = cluster["workload"]
             if "transactions" in workload:
-                for k, v in workload["transactions"].items():
-                    self.maybe_gauge("foundationdb.workload.transactions." + k + ".hz", v, "hz")
-                    self.maybe_count("foundationdb.workload.transactions." + k + ".counter", v, "counter")
+                transactions = workload["transactions"]
+                for k in transactions:
+                    self.maybe_hz_counter("foundationdb.workload.transactions." + k, transactions, k)
 
             if "operations" in workload:
-                for k, v in workload["operations"].items():
-                    self.maybe_gauge("foundationdb.workload.operations." + k + ".hz", v, "hz")
-                    self.maybe_count("foundationdb.workload.operations." + k + ".counter", v, "counter")
+                operations = workload["operations"]
+                for k in operations:
+                    self.maybe_hz_counter("foundationdb.workload.operations." + k, operations, k)
 
         if "latency_probe" in cluster:
             for k, v in cluster["latency_probe"].items():
                 self.gauge("foundationdb.latency_probe." + k, v)
 
         self.service_check("foundationdb.can_connect", AgentCheck.OK)
 
-    def maybe_gauge(self, metric, obj, key):
+    def maybe_gauge(self, metric, obj, key, tags=None):
+        if key in obj:
+            self.gauge(metric, obj[key], tags=tags)
+
+    def maybe_count(self, metric, obj, key, tags=None):
         if key in obj:
-            self.gauge(metric, obj[key])
+            self.count(metric, obj[key], tags=tags)
 
-    def maybe_count(self, metric, obj, key):
+    def maybe_hz_counter(self, metric, obj, key, tags=None):
         if key in obj:
-            self.count(metric, obj[key])
+            if "hz" in obj[key]:
+                self.gauge(metric + ".hz", obj[key]["hz"], tags=tags)
+            if "counter" in obj[key]:
+                self.count(metric + ".counter", obj[key]["counter"], tags=tags)
```

Comment:
Hmm, actually `count` may be wrong here because it's documented as:

> This function submits the number of events that occurred during the check interval. It can be called multiple times during a check’s execution, each sample being added to the value that is sent.

Whereas the number that FoundationDB gives us is in "all time" (well, probably since it was started). I think we want `monotonic_count`, described as:

> This function is used to track a raw COUNT metric that always increases. The Datadog Agent calculates the delta between each submission. Samples that have a lower value than the previous sample are ignored. Lower values usually indicate the underlying raw COUNT metric has been reset.